src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java

```diff
@@ -0,0 +1,40 @@
+package edu.wpi.grip.core.operations.opencv;
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.Operation;
+import edu.wpi.grip.core.Socket;
+import edu.wpi.grip.core.SocketHint;
+import org.bytedeco.javacpp.opencv_core;
+import org.bytedeco.javacpp.opencv_core.*;
+
+
+/**
+ * Performs the opencv add operation
+ */
+public class AddOperation implements Operation {
+    private SocketHint<Mat>
+            aHint = new SocketHint<>("a", Mat.class),
+            bHint = new SocketHint<>("b", Mat.class),
+            sumHint = new SocketHint<>("sum", Mat.class);
+
+    @Override
+    public Socket[] createInputSockets(EventBus eventBus) {
+        return new Socket[]{new Socket<>(eventBus, aHint), new Socket<>(eventBus, bHint)};
+    }
+
+    @Override
+    public Socket[] createOutputSockets(EventBus eventBus) {
+        System.out.println("entering createOutputSockets");
+        Socket<Mat> output = new Socket<>(eventBus, sumHint);
+        System.out.println("middle createOutputSockets");
+        output.setValue(new Mat());
```

Comment:
There's no need to call `setValue()`.  `SocketHint` has a constructor that takes an initial value.